pydrive2/test/test_file.py

```diff
@@ -25,6 +28,59 @@
 )
 
 
+def auth_with_resource_key_mock() -> GoogleAuth:
+    """
+    Create GoogleAuth with mocked inner httplib2.Http simulating need
+    for resourceKey header.
+    """
+    http_mock = MagicMock()
+
+    def resource_key_request(
+        uri,
+        method="GET",
+        body=None,
+        headers=None,
+        redirections=1,
+        connection_type=None,
+    ):
+        """httplib2.Http.request mock."""
+        # If "media" is not requested, it means we expect metadata
+        fetch_meta_data = "alt=media" not in uri
```

Comment:
how do those `uri`s look like in both case? can we actually distinguish by path?

Reply:
Sadly it's the same URL (files endpoint); only the query parameters differ, thus my choice to rely on the media one which is discriminant.